evm/contracts/OBridge.sol

```diff
@@ -0,0 +1,324 @@
+        return msg.data;
+    }
+}
+
+abstract contract Ownable is Context {
+    address public owner;
+    address public nextOwner;
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    /**
+     * @dev Initializes the contract setting the deployer as the initial owner.
+     */
+    constructor() {
+        _setOwner(_msgSender());
+    }
+
+    /**
+     * @dev Throws if called by any account other than the owner.
+     */
+    modifier onlyOwner() {
+        require(owner == _msgSender(), "Ownable: caller is not the owner");
+        _;
+    }
+
+    /**
+     * @dev Leaves the contract without owner. It will not be possible to call
+     * `onlyOwner` functions anymore. Can only be called by the current owner.
+     *
+     * NOTE: Renouncing ownership will leave the contract without an owner,
+     * thereby removing any functionality that is only available to the owner.
+     */
+    function renounceOwnership() public virtual onlyOwner {
+        _setOwner(address(0));
+    }
+
+    /**
+    * @dev Allows the current owner to transfer control of the contract to a newOwner.
+    * @param _newOwner The address to transfer ownership to.
+    */
+    function transferOwnership(address _newOwner) public onlyOwner {
+        require(_newOwner != address(0), "Address should not be 0x");
+        nextOwner = _newOwner;
+    }
+    
+    function approveOwnership() public{
+        require(nextOwner == msg.sender);
+        owner = nextOwner;
+    }
+
+    function _setOwner(address newOwner) private {
+        address oldOwner = owner;
+        owner = newOwner;
+        emit OwnershipTransferred(oldOwner, newOwner);
+    }
+}
+
+contract BridgeFee is Ownable{
+    uint256 public basisPointsRate = 0;
+    mapping(address => uint256) public maximumFee;
+    address public tollAddress;
+
+    /**
+     * @dev Initializes the contract setting the deployer as the initial owner.
+     */
+    constructor() {
+        tollAddress = _msgSender();
+    }
+
+    function setBasisPointsRate(uint256 rate) external onlyOwner{
+        basisPointsRate = rate;
+    }
+
+    function setMaximumFee(address token, uint256 fee) external onlyOwner{
+        maximumFee[token] = fee;
+    }
+
+    function setTollAddress(address toll) external onlyOwner {
+        tollAddress = toll;
+    }
+
+    function calcFee(address token, uint256 value) view internal returns (uint256) {
+        uint256 fee = value * basisPointsRate / 10000;
+
+        uint256 maxFee = maximumFee[token];
+        if (maxFee > 0 && fee > maxFee){
+            fee = maxFee;
+        }
+
+        return fee;
+    }
+}
+
+contract OBridge is BridgeFee{
+    using SafeERC20 for IERC20;
+
+    enum TransferStatus {
+        Null,
+        Pending,
+        Confirmed,
+        Refunded
+    }
+
+    mapping(bytes32 => TransferStatus) public transfers;
+
+    event LogNewTransferOut(
+        bytes32 transferId,
+        address sender,
```

Comment:
I thought we want yo make the address generic (eg uint256) since some addresses might be from non-EVM chains?

Reply:
The next submission will modify the payment address to uint256. But it is not the sender here

Reply:
https://github.com/obridge-io/atomic-swap-contracts/commit/51541be2ee04bff5c7b27f4c51da2caf7f5df44a